bootstrap/kubeadm/api/v1alpha3/zz_generated.conversion.go

```diff
@@ -384,7 +384,34 @@ func Convert_v1alpha4_KubeadmConfigList_To_v1alpha3_KubeadmConfigList(in *v1alph
 }
 
 func autoConvert_v1alpha3_KubeadmConfigSpec_To_v1alpha4_KubeadmConfigSpec(in *KubeadmConfigSpec, out *v1alpha4.KubeadmConfigSpec, s conversion.Scope) error {
-	out.ClusterConfiguration = (*v1alpha4.ClusterConfiguration)(unsafe.Pointer(in.ClusterConfiguration))
+	if in.ClusterConfiguration != nil {
+		out := &out.ClusterConfiguration
+		in := in.ClusterConfiguration
+		*out = new(v1alpha4.ClusterConfiguration)
+		(*out).DNS = v1alpha4.DNS{
+			ImageMeta: v1alpha4.ImageMeta{
+				ImageRepository: in.DNS.ImageRepository,
+				ImageTag:        in.DNS.ImageTag,
+			},
+		}
+		(*out).APIServer = *(*v1alpha4.APIServer)(unsafe.Pointer(&in.APIServer))
+		(*out).APIVersion = in.APIVersion
+		(*out).CertificatesDir = in.CertificatesDir
+		(*out).ClusterName = in.ClusterName
+		(*out).ControlPlaneEndpoint = in.ControlPlaneEndpoint
+		(*out).ControllerManager = *(*v1alpha4.ControlPlaneComponent)(unsafe.Pointer(&in.ControllerManager))
+		(*out).Etcd = *(*v1alpha4.Etcd)(unsafe.Pointer(&in.Etcd))
+		(*out).FeatureGates = in.FeatureGates
+		(*out).ImageRepository = in.ImageRepository
+		(*out).Kind = in.Kind
+		(*out).KubernetesVersion = in.KubernetesVersion
+		(*out).Networking = *(*v1alpha4.Networking)(unsafe.Pointer(&in.Networking))
+		(*out).Scheduler = *(*v1alpha4.ControlPlaneComponent)(unsafe.Pointer(&in.Scheduler))
+		(*out).TypeMeta = in.TypeMeta
+		(*out).UseHyperKubeImage = in.UseHyperKubeImage
+	} else {
+		out.ClusterConfiguration = nil
+	}
```

Comment:
@fabriziopandini  @vincepri not sure about this part, I'm quite sure it can be improved but I didn't found a better way to handle conversion